doc/toscalite/spec.md

```diff
@@ -0,0 +1,169 @@
+# TOSCAlite specification
+
+## Introduction
+
+TOSCAlite is a very lightweight version of the OASIS TOSCA standard. It can be used to model a deployment based on Linux systems (mostly Ubuntu 16.04 LTS) 
+
+## Nodes
+
+A node defined in the ``model.xml`` (see Packaging) represents a node in a TOSCAlite topology. 
+TOSCAlite defines two types of nodes. It is not possible to extend TOSCAlite with more types of nodes.
+
+In XML a node gets defined as following:
+
+```XML
+<Node>
+    <Type>type</Type>
+    <Name>name</Name>
+    <Properties>
+        <Property key="keyname">value</Property>
+    </Properties>
+</Node>
+```
+**Explanation:**
+* ``Type`` - This defines the node type. In TOSCAlite only two types are allowed. (``service`` and ``machine``)
+* ``Ǹame`` - The name of a node is used to uniquely identify the node in the topology. It is the nodes primary identifier. The names are only allowed to contain the following characters ``abcdefghijklmnopqrstuvwxyz-_1234567890``
+* ``Properties`` -  Has to contain Property elements, with the attribute Key. the key is required to be a lower-case name without spaces (allowed characters `abcdefghijklmnopqrstuvxyz`). 
+
+The ``Type`` and ``Name`` elements are always required. ``Properties`` can be optional, at least in service nodes.
+
+### Machines
+
+A machine node represents a Linux-based system, probably Ubuntu 16.04. The deployment system does not care where the system is running, as long as the machine is reachable with SSH.
```

Comment:
Maybe stress that ssh is mandatory! 
I think it might be better to say the machine is reachable and ssh is installed and properly configured.
Because without ssh nothing will work :smile: